stores/useWalletStore.tsx

```diff
@@ -29,54 +42,227 @@ interface WalletStore extends State {
+        getAccountTypes(Realm)
       )
 
-      console.log('fetchProposals', proposals)
+      set((s) => {
+        s.realms = realms
+      })
 
-      set((state) => {
-        state.proposals = proposals
+      const mints = await Promise.all(
+        Object.values(realms).map((r) =>
+          getMint(connection, r.info.communityMint)
+        )
+      )
+
+      set((s) => {
+        s.mints = Object.fromEntries(
+          mints.map((m) => [m.publicKey.toBase58(), m.account])
+        )
+      })
+    },
+    async fetchRealm(programId: PublicKey, realmId: PublicKey) {
+      const endpoint = get().connection.endpoint
+      const realms = get().realms
+      const mints = get().mints
+      const realm = realms[realmId.toBase58()]
+      const realmMintPk = realm.info.communityMint
+      const realmMint = mints[realmMintPk.toBase58()]
+      const set = get().set
+
+      console.log('fetchRealm', programId.toBase58(), realmId.toBase58())
+
+      const [governances, tokenRecords] = await Promise.all([
+        getGovernanceAccounts<Governance>(
+          programId,
+          endpoint,
+          Governance,
+          getAccountTypes(Governance),
+          [pubkeyFilter(1, realmId)]
+        ),
+
+        getGovernanceAccounts<TokenOwnerRecord>(
+          programId,
+          endpoint,
+          TokenOwnerRecord,
+          getAccountTypes(TokenOwnerRecord),
+          [pubkeyFilter(1, realmId), pubkeyFilter(1 + 32, realmMintPk)]
+        ),
+      ])
+
+      const tokenRecordsByOwner = mapEntries(tokenRecords, ([_k, v]) => [
+        v.info.governingTokenOwner.toBase58(),
+        v,
+      ])
+
+      console.log('fetchRealm mint', realmMint)
+      console.log('fetchRealm governances', governances)
+      console.log('fetchRealm tokenRecords', tokenRecordsByOwner)
+
+      set((s) => {
+        s.selectedRealm.realm = realm
+        s.selectedRealm.mint = realmMint
+        s.selectedRealm.governances = governances
+        s.selectedRealm.tokenRecords = tokenRecordsByOwner
+      })
+
+      const proposalsByGovernance = await Promise.all(
+        mapKeys(governances, (g) => {
+          return getGovernanceAccounts<Proposal>(
+            programId,
+            endpoint,
+            Proposal,
+            getAccountTypes(Proposal),
+            [pubkeyFilter(1, new PublicKey(g))]
+          )
+        })
+      )
+
+      const proposals = merge(...proposalsByGovernance)
+
+      console.log('fetchRealm proposals', proposals)
+
+      set((s) => {
+        s.selectedRealm.proposals = proposals
+      })
+
+      const votesByProposal = await Promise.all(
+        mapKeys(proposals, async (p) => {
+          const voteRecords = await getGovernanceAccounts<VoteRecord>(
+            programId,
+            endpoint,
+            VoteRecord,
+            getAccountTypes(VoteRecord),
+            [pubkeyFilter(1, new PublicKey(p))]
+          )
+
+          const precision = Math.pow(10, 6)
+          const yes =
```

Comment:
There is no need to manually calculate yes/no votes. You have `yesVotesCount` and `noVotesCount` on `Proposal`

Reply:
good catch! removed it